src/payment/models/pricing/utils/price-step.ts

```diff
@@ -0,0 +1,165 @@
+import { cloneDeep } from 'lodash';
+import { Price } from '../../exchange/dto/price.dto';
+import { PriceMismatchException } from '../../exchange/exceptions/price-mismatch.exception';
+import { PriceStepResult, PriceProvider, PriceProviderName } from '../interfaces';
+import { PriceStepInitSpecification } from '../specifications/price-step-init.specification';
+
+export interface PriceStepOptions {
+  from?: string | 'input';
+  to?: string | 'output';
+  referenceTo?: string;
+  providers?: PriceStepProviders;
+  fixedPrice?: number;
+}
+
+export interface PriceStepProviders {
+  primary: PriceProvider[];
+  reference: PriceProvider[];
+}
+
+export class PriceStep {
+  private readonly options: PriceStepOptions = {};
+
+  constructor(options: PriceStepOptions) {
+    this.options.from = options.from || 'input';
+    this.options.to = options.to || 'output';
+    this.options.referenceTo = options.referenceTo;
+    this.options.providers = {
+      primary: options.providers?.primary || [],
+      reference: options.providers?.reference || [],
+    };
+    this.options.fixedPrice = options.fixedPrice;
```

Comment:
Why not like this (only initialize the options in constructor)?
```ts
this.options = {
   from: options.from || 'input',
   to: options.to || 'output',
   ...
};
```

Reply:
totally